### ISSUE TITLE
Unwrapping - Phass not found and files too big for snaphu

### ISSUE THREAD

**Issue by jlmaurer** [isce-framework/fringe#21]
- When running the unwrapping script with snaphu I get the message:
   - "one or more interferogram dimensions too large"
   - file size is (67557.0,18929.0)
- Phass doesn't seem to exist (can't be imported)
  - Is it another library?

**Comment by hfattahi** [comment]
Phass is another unwrapping software. Currently it's not available.

**Comment by piyushrpt** [comment]
The use case for this tool might not be to a wide-area analysis at high resolution in a single shot. For wide area analysis, if you are still interested in high-res; we will need a tiling mechanism that puts together results after unwrapping. I think that will come much later. If you are interested in moderate/low resolution results over a large area - you might be better off multi-looking fringe results and performing analysis with that.

**Comment by jlmaurer** [comment]
Yes, that's what I was thinking. Sounds good.

**Comment by scottstanie** [comment]
I got the same  "dimensions too large" error from processing just a .5 degree x .5 degree block, which came out to have interferograms with shape (7335, 45053). For me, giving the `--tile` option to snaphu avoided the error and unwrapped successfully

I'm not sure if it's available as the `Snaphu` component that `unwrap_fringe.py` uses, since i've only used configuration files/ command line calls to snaphu before, but I'll attach the config I used in case @jlmaurer want's to try again with snaphu. For reference, my interferogram unwrapped in 14 minutes with the tile settings I had, but it was on a pretty big machine.

```bash
$ cat snaphu1.conf
########################
# Unwrapping parameters
#########################

STATCOSTMODE SMOOTH
INITMETHOD MCF
VERBOSE TRUE

###############
# Input files
###############

CORRFILE tcorr_ds_ps.bin

################
# Output files
################

OUTFILE unwrap/20160930_20170104.unw
LOGFILE snaphu.log

################
# File formats
################

# INFILEFORMAT FLOAT_DATA
CORRFILEFORMAT FLOAT_DATA
# OUTFILEFORMAT FLOAT_DATA


################
# Tile control
################

NTILEROW 4
NTILECOL 12

ROWOVRLP 30
COLOVRLP 30

NPROC 48
```

**Comment by jlmaurer** [comment]
Thanks @scottstanie I'll try that option out!

**Comment by stoormgeo** [comment]
> I got the same "dimensions too large" error from processing just a .5 degree x .5 degree block, which came out to have interferograms with shape (7335, 45053). For me, giving the `--tile` option to snaphu avoided the error and unwrapped successfully
> 
> I'm not sure if it's available as the `Snaphu` component that `unwrap_fringe.py` uses, since i've only used configuration files/ command line calls to snaphu before, but I'll attach the config I used in case @jlmaurer want's to try again with snaphu. For reference, my interferogram unwrapped in 14 minutes with the tile settings I had, but it was on a pretty big machine.
> 
> ```shell
> $ cat snaphu1.conf
> ########################
> # Unwrapping parameters
> #########################
> 
> STATCOSTMODE SMOOTH
> INITMETHOD MCF
> VERBOSE TRUE
> 
> ###############
> # Input files
> ###############
> 
> CORRFILE tcorr_ds_ps.bin
> 
> ################
> # Output files
> ################
> 
> OUTFILE unwrap/20160930_20170104.unw
> LOGFILE snaphu.log
> 
> ################
> # File formats
> ################
> 
> # INFILEFORMAT FLOAT_DATA
> CORRFILEFORMAT FLOAT_DATA
> # OUTFILEFORMAT FLOAT_DATA
> 
> 
> ################
> # Tile control
> ################
> 
> NTILEROW 4
> NTILECOL 12
> 
> ROWOVRLP 30
> COLOVRLP 30
> 
> NPROC 48
> ```


> Thanks @scottstanie I'll try that option out!

@scottstanie @jlmaurer  
Did you get the correct unwrapped images after using the tile mode?
My issue https://github.com/isce-framework/fringe/issues/46

**Comment by scottstanie** [comment]
I've had relatively few problems with the tile mode of snaphu. Only recently did I have an interferogram where there was an obvious tile artifact.
BTW, the most recent snaphu gives this warning:

```bash
$ /home/scott/phase_unwrap/bin/snaphu -s ./20150706_20160302.int 3120 -c ./20150706_20160302.cc -o ./20150706_20160302.unw -S --tile 3 3 30 30 --nproc 9

snaphu v2.0.4
WARNING: Tile overlap is small (may give bad results)
Starting first-round tile-mode unwrapping
...
```

Apparently my "30 30" overlaps were less than it was expecting. If you give larger overlaps (like in the 300- 500 range), it stops giving that warning. With less overlap, snaphu has less data to figure out the across-tile solutions at the end.

But I'll say that the tiling has never given a result like the issue you linked to. That diagonally-repeating binary messed up output almost always comes from my giving the display program the wrong file width, or from saving it using the wrong width.

**Comment by hfattahi** [comment]
> I've had relatively few problems with the tile mode of snaphu. Only recently did I have an interferogram where there was an obvious tile artifact.
> BTW, the most recent snaphu gives this warning:
> 
> ```shell
> $ /home/scott/phase_unwrap/bin/snaphu -s ./20150706_20160302.int 3120 -c ./20150706_20160302.cc -o ./20150706_20160302.unw -S --tile 3 3 30 30 --nproc 9
> 
> snaphu v2.0.4
> WARNING: Tile overlap is small (may give bad results)
> Starting first-round tile-mode unwrapping
> ...
> ```
> 
> Apparently my "30 30" overlaps were less than it was expecting. If you give larger overlaps (like in the 300- 500 range), it stops giving that warning. With less overlap, snaphu has less data to figure out the across-tile solutions at the end.
> 
> But I'll say that the tiling has never given a result like the issue you linked to. That diagonally-repeating binary messed up output almost always comes from my giving the display program the wrong file width, or from saving it using the wrong width.

Yes definitely larger overlap reduces the possibility of tile boundary artifacts. I agree the plot looks to me as a display issue. That's why I was asking about the shape and size of the outputs in https://github.com/isce-framework/fringe/issues/46 .

**Comment by Jaapel** [comment]
Hi all, I also encountered this issue.
The `snaphu.h` file defines `LARGESHORT` as 32000. If one your dimensions is larger than 32000, this error will be raised when reading any input file in function `ReadInputFile`.
